x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/EqlPlugin.java

```diff
@@ -103,9 +102,24 @@
     public List<Setting<?>> getSettings() {
```

Comment:
Either we use `enabled` or this method. On one hand `enabled` checks the property presence statically but this method tries to do it dynamic when a class is instantiated.
There's there are multiple pieces of `isSnapshot` in this class that need simplification.

Reply:
tried to follow the autoscaling plugin pattern, that I was told should be an example of handling the feature flag.